src/brevitas/graph/quantize.py

```diff
@@ -0,0 +1,598 @@
+                    continue
+
+                # Generate a QuantIdentity module to use for alignement of the inputs
+                shared_quant_identity, shared_quant_identity_name = _get_quant_module(
+                    model, node, quant_identity_map, quant_act_map, unsigned_act_tuple)
+
+                # Recursively, for every input node, traverse the graph to determine how to quantize
+                # and align that input node.
+                recursive_input_handler(
+                    model,
+                    node,
+                    shared_quant_identity_name,
+                    shared_quant_identity,
+                    rewriters,
+                    quant_identity_map,
+                    align_input_quant_fn,
+                    align_sign=same_sign)
+                for rewriter in rewriters:
+                    model = rewriter.apply(model)
+                model.graph.lint()
+                model.recompile()
+                return False
+        return True
+
+    while not is_converged(model):
+        continue
+
+    return model
+
+
+def add_output_quant_handler(model, quant_identity_map, quant_act_map, unsigned_act_tuple):
+    """
+    Check the output of every add node, to decide whether it needs to be requantized or not, based
+    on the logic of output_quant_handler.
+    """
+    rewriters = []
+    for node in model.graph.nodes:
+        if (node.op == 'call_function' and node.target in ADD_FNS or
+                node.op == 'call_method' and node.target in ADD_METHODS):
+            output_quant_handler(
+                model,
+                node,
+                rewriters,
+                is_sign_preserving=True,
+                quant_identity_map=quant_identity_map,
+                quant_act_map=quant_act_map,
+                unsigned_act_tuple=unsigned_act_tuple)
+    for rewriter in rewriters:
+        model = rewriter.apply(model)
+    return model
+
+
+def wbiol_handler(model, float_to_quant_wbiol_map, quant_identity_map):
+    """
+    Replace FP weight layers with their corresponding quantized version
+    """
+    rewriters = []
+    for node in model.graph.nodes:
+        if node.op == 'call_module':
+            module = get_module(model, node.target)
+            if isinstance(module, tuple(float_to_quant_wbiol_map.keys())):
+                output_quant_handler(
+                    model,
+                    node,
+                    rewriters,
+                    is_sign_preserving=False,
+                    quant_identity_map=quant_identity_map)
+                quant_module_class, quant_module_kwargs = float_to_quant_wbiol_map[type(module)]
+                rewriter = ModuleToModuleByInstance(
+                    module, quant_module_class, **quant_module_kwargs)
+                rewriters.append(rewriter)
+    for rewriter in rewriters:
+        model = rewriter.apply(model)
+    return model
+
+
+def avgpool_handler(
```

Comment:
add a flag requantize_output, that is then exposed at the top of quantize as requantize_avgpool_output